metpy/calc/thermo.py

```diff
@@ -354,6 +354,9 @@ def lfc(pressure, temperature, dewpt, parcel_temperature_profile=None):
             x, y = lcl(pressure[0], temperature[0], dewpt[0])
             return x, y
     else:
+        idx = x < lcl(pressure[0], temperature[0], dewpt[0])[0]
+        x = x[idx]
```

Comment:
If you wrote this using:
```python
idx = np.argmax(x < lcl(...), axis=0)
idx = metpy.calc.tools.broadcast_indices(...)
```
this logic at least would work on n-dimensional grids. If you prefer not to deal with it right now, that’s fine.

Reply:
Given that the function itself isn't n-dim ready, probably not yet.